server/src/main/java/keywhiz/service/providers/ClientAuthFactory.java

```diff
@@ -134,12 +135,23 @@ public Client provide(ContainerRequest containerRequest,
   }
 
   private Client authorizeClientFromXfccHeader(XfccSourceConfig xfccConfig,
```

Comment:
nit: Since the XFCC header isn't necessarily what's used, maybe this should be renamed to `authoritizeClientFromForwardedData` or something.

Reply:
(though this would be a bigger change, and should probably be handled in a separate commit, since the term "XFCC" is used a lot in this code)

Reply:
I think it's fine to keep XFCC name because this custom header is an optional config inside XFCC, just to support a scenario where a real client is behind a proxy.